app/javascript/controllers/complex_radio_controller.js

```diff
@@ -11,7 +11,7 @@ export default class extends Controller {
     this.selectionTargets.forEach((target) => {
       const checked = target.querySelector("input[type='radio']").checked
       // If radio is checked, enable its child select elements; if unchecked, disable.
-      target.querySelectorAll('select,input[type="number"],input[type="checkbox"]').forEach((element) => element.disabled = !checked)
+      target.querySelectorAll('select,input[type="number"],input[type="checkbox"],fieldset').forEach((element) => element.disabled = !checked)
```

Comment:
Cool. I didn't know a fieldset could be disabled: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset#attr-disabled